lib/puppet/indirector/facts/facter.rb

```diff
@@ -83,6 +88,21 @@ def self.setup_external_search_paths(request)
       dirs << dir
     end
 
+    dirs << request.options[:external_dir] if request.options[:external_dir]
     Facter.search_external dirs
   end
+
+  private
+
+  def find_with_options(request)
+    options = request.options
+    options_for_facter = String.new
+    options_for_facter += options[:user_query].join(' ')
+    options_for_facter += " --config #{options[:config_file]}" if options[:config_file]
+    options_for_facter += " --show-legacy" if options[:show_legacy]
+    options_for_facter += " --no-block" if options[:no_block] == false
+    options_for_facter += " --no-cache" if options[:no_cache] == false
+
+    Puppet::Node::Facts.new(request.key, Facter.resolve(options_for_facter))
```

Comment:
Looks good, could you add a check that the Facter responds to `:resolve` or bump puppet's minimum facter dependency, something like `['>= 4.0.whatever', '< 5.0']`?

Reply:
I added a check for Facter to respond to :resolve